open_event/helpers/data_getter.py

```diff
@@ -178,10 +178,10 @@ def get_sessions_of_user(upcoming_events=True):
         """
         if upcoming_events:
             return Session.query.filter(Session.speakers.any(Speaker.user_id == login.current_user.id)).filter(
-                Event.state != 'Completed')
+                Session.start_time >= datetime.datetime.now())
         else:
             return Session.query.filter(Session.speakers.any(Speaker.user_id == login.current_user.id)).filter(
-                Event.state == 'Completed')
```

Comment:
The previous logic was correct. It is supposed to be sessions in upcoming **events** and past **events**. See the original issue #703 .